Sources/Sentry/SentryClient.m

```diff
@@ -249,10 +249,19 @@ - (SentryEvent *)buildErrorEvent:(NSError *)error
     // as a list of exceptions with error mechanisms, sorted oldest to newest (so, the leaf node
     // underlying error as oldest, with the root as the newest)
     NSMutableArray<NSError *> *errors = [NSMutableArray<NSError *> arrayWithObject:error];
-    NSError *underlyingError = error.userInfo[NSUnderlyingErrorKey];
+    NSError *underlyingError;
+    if ([error.userInfo[NSUnderlyingErrorKey] isKindOfClass:[NSError class]]) {
+        underlyingError = error.userInfo[NSUnderlyingErrorKey];
+    }
+
     while (underlyingError != nil) {
         [errors addObject:underlyingError];
-        underlyingError = underlyingError.userInfo[NSUnderlyingErrorKey];
+
+        if ([underlyingError.userInfo[NSUnderlyingErrorKey] isKindOfClass:[NSError class]]) {
+            underlyingError = underlyingError.userInfo[NSUnderlyingErrorKey];
+        } else {
+            underlyingError = nil;
```

Comment:
We should consider logging and/or client reporting for invalid underlying errors.

Reply:
Good call. `SENTRY_LOG_*` looked like the most common logging pattern so I continued with that.